src/components/Api.js

```diff
@@ -0,0 +1,212 @@
+import Section from './Section.js';
+import {createNewCard} from '../utils/utils.js';
+import {cardContainerClass} from '../utils/constants.js';
+import { userInfo, formValidatorCardModal } from '../pages/index.js';
+
+export default class Api{
+    constructor(){
+      this._baseUrl = 'https://around-api.en.tripleten-services.com/v1',
+      this._headers = {
+        authorization: "4c45d989-e1aa-4bb6-a467-6ac9c46f3dac",
+        "Content-Type": "application/json"
+      };
+      this._headers1 = {
+        authorization: "4c45d989-e1aa-4bb6-a467-6ac9c46f3dac",
+      }
+    }
+
+    loadUserInfo(){
+      return fetch(`${this._baseUrl}/users/me`, {
+        headers: this._headers1
+      })
+      .then(res => {
+        if(res.ok){
+          return res.json();
+        } else{
+          return Promise.reject(`Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the `reference` for the method rather than the call. 

**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.